internal/services/web/app_service_certificate_resource.go

```diff
@@ -166,10 +173,8 @@ func resourceAppServiceCertificateCreateUpdate(d *pluginsdk.ResourceData, meta i
 		Tags:     tags.Expand(t),
 	}
 
-	if len(hostingEnvironmentProfileId) > 0 {
-		certificate.CertificateProperties.HostingEnvironmentProfile = &web.HostingEnvironmentProfile{
-			ID: &hostingEnvironmentProfileId,
-		}
+	if appServicePlanId != "" {
+		certificate.CertificateProperties.ServerFarmID = &appServicePlanId
```

Comment:
Should we still be setting `hosting_environment_profile_id` together with the new option until 3.0?

Reply:
I think this is still a valid computed attribute. Its filled with the app-service-environment ID. See REST API: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/web/resource-manager/Microsoft.Web/stable/2020-06-01/Certificates.json#L395